### ISSUE TITLE
Calculation Mistake

### ISSUE THREAD

**Issue by redStarP2** [fkleon/math-expressions#40]
Mycode;
```
print("(0.7 / 100)");
exp = p.parse("(0.7 / 100)");
print(exp.simplify().evaluate(EvaluationType.REAL, null));
```
Output;
```
0.006999999999999999
```

Expected;
```
0.007
```
![wtf](https://user-images.githubusercontent.com/24715883/103442137-95e22b00-4c64-11eb-88b1-7e04c28720ae.png)

is there anything I'm doing wrong ?

**Comment by fkleon** [comment]
Hi @redStarP2, thanks for the report.

This is expected as this library is using double values internally, and Dart is using 64-bit floating point numbers as specified by IEEE 754 (see the [DartDoc for double](https://api.dart.dev/stable/2.10.4/dart-core/double-class.html)). Note that this is correct behaviour and for example JavaScript and Java behaving the same way.

A good explanation with a suggested workaround can be found [here on Stackoverflow](https://stackoverflow.com/questions/57481767/dart-rounding-errors).

In summary, to avoid this you need to round the result to a precision of your choice, e.g.

```dart
final exp = p.parse("(0.7 / 100)");
final result = exp.evaluate(EvaluationType.REAL, null);

final precision = 1000;
final resultRoundedToPrecision = (result * precision).round() / precision;

print(result);
print(resultRoundedToPrecision);
```

Output:
```
0.006999999999999999
0.007
```
